src/coreclr/src/jit/codegenxarch.cpp

```diff
@@ -6294,6 +6294,22 @@ void CodeGen::genCompareInt(GenTree* treeNode)
     instruction ins;
     var_types   type = TYP_UNKNOWN;
 
+#ifdef TARGET_64BIT
```

Comment:
Why is this 64-bit only?

Reply:
I don't remember 🙂 but I've checked - the opt works fine in 32bits.
Also, I've added `x>=0` case, now I'm re-running jit-diff.
also checking if we correctly morph various `x>-1` to `x>=0` etc